@bdbaddog That seems to work well for `libmodules.a`, thanks a lot!

I still get the same linking issue eventually as I do with this workaround PR, not sure yet why:
```
g++ -o bin/godot.windows.opt.tools.64.exe -static platform/windows/godot_windows.windows.opt.tools.64.o platform/windows/context_gl_windows.windows.opt.tools.64.o platform/windows/crash_handler_windows.windows.opt.tools.64.o platform/windows/os_windows.windows.opt.tools.64.o platform/windows/key_mapping_windows.windows.opt.tools.64.o platform/windows/joypad_windows.windows.opt.tools.64.o platform/windows/power_windows.windows.opt.tools.64.o platform/windows/windows_terminal_logger.windows.opt.tools.64.o platform/windows/godot_res.windows.opt.tools.64.o main/libmain.windows.opt.tools.64.a main/tests/libtests.windows.opt.tools.64.a modules/libmodules.windows.opt.tools.64.a platform/libplatform.windows.opt.tools.64.a drivers/libdrivers.windows.opt.tools.64.a editor/libeditor.windows.opt.tools.64.a scene/libscene.windows.opt.tools.64.a servers/libservers.windows.opt.tools.64.a core/libcore.windows.opt.tools.64.a modules/freetype/libfreetype_builtin.windows.opt.tools.64.a -lmingw32 -lopengl32 -ldsound -lole32 -ld3d9 -lwinmm -lgdi32 -liphlpapi -lshlwapi -lwsock32 -lws2_32 -lkernel32 -loleaut32 -ldinput8 -ldxguid -lksuser -limm32 -lbcrypt -lavrt -luuid -ldwmapi
=====
b"C:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: modules/libmodules.windows.opt.tools.64.a(register_module_types.gen.windows.opt.tools.64.o): in function `register_module_types()':\r\nC:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:53: undefined reference to `register_assimp_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:56: undefined reference to `register_bmp_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:59: undefined reference to `register_bullet_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:62: undefined reference to `register_camera_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:65: undefined reference to `register_csg_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:68: undefined reference to `register_cvtt_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:71: undefined reference to `register_dds_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:74: undefined reference to `register_enet_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:77: undefined reference to `register_etc_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:80: undefined reference to `register_freetype_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:83: undefined reference to `register_gdnative_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:86: undefined reference to `register_gdscript_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:89: undefined reference to `register_gridmap_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:92: undefined reference to `register_hdr_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:95: undefined reference to `register_jpg_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:98: undefined reference to `register_jsonrpc_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:101: undefined reference to `register_mbedtls_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:104: undefined reference to `register_mobile_vr_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:110: undefined reference to `register_ogg_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:113: undefined reference to `register_opensimplex_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:116: undefined reference to `register_opus_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:119: undefined reference to `register_pvr_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:122: undefined reference to `register_recast_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:125: undefined reference to `register_regex_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:128: undefined reference to `register_squish_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:131: undefined reference to `register_stb_vorbis_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:134: undefined reference to `register_svg_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:137: undefined reference to `register_tga_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:140: undefined reference to `register_theora_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:143: undefined reference to `register_tinyexr_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:146: undefined reference to `register_upnp_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:149: undefined reference to `register_vhacd_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:152: undefined reference to `register_visual_script_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:155: undefined reference to `register_vorbis_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:158: undefined reference to `register_webm_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:161: undefined reference to `register_webp_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:164: undefined reference to `register_webrtc_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:167: undefined reference to `register_websocket_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: modules/libmodules.windows.opt.tools.64.a(register_module_types.gen.windows.opt.tools.64.o): in function `unregister_module_types()':\r\nC:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:180: undefined reference to `unregister_assimp_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:183: undefined reference to `unregister_bmp_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:186: undefined reference to `unregister_bullet_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:189: undefined reference to `unregister_camera_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:192: undefined reference to `unregister_csg_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:195: undefined reference to `unregister_cvtt_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:198: undefined reference to `unregister_dds_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:201: undefined reference to `unregister_enet_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:204: undefined reference to `unregister_etc_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:207: undefined reference to `unregister_freetype_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:210: undefined reference to `unregister_gdnative_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:213: undefined reference to `unregister_gdscript_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:216: undefined reference to `unregister_gridmap_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:219: undefined reference to `unregister_hdr_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:222: undefined reference to `unregister_jpg_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:225: undefined reference to `unregister_jsonrpc_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:228: undefined reference to `unregister_mbedtls_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:231: undefined reference to `unregister_mobile_vr_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:237: undefined reference to `unregister_ogg_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:240: undefined reference to `unregister_opensimplex_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:243: undefined reference to `unregister_opus_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:246: undefined reference to `unregister_pvr_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:249: undefined reference to `unregister_recast_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:252: undefined reference to `unregister_regex_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:255: undefined reference to `unregister_squish_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:258: undefined reference to `unregister_stb_vorbis_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:261: undefined reference to `unregister_svg_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:264: undefined reference to `unregister_tga_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:267: undefined reference to `unregister_theora_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:270: undefined reference to `unregister_tinyexr_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:273: undefined reference to `unregister_upnp_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:276: undefined reference to `unregister_vhacd_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:279: undefined reference to `unregister_visual_script_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:282: undefined reference to `unregister_vorbis_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:285: undefined reference to `unregister_webm_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:288: undefined reference to `unregister_webp_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:291: undefined reference to `unregister_webrtc_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/modules/register_module_types.gen.cpp:294: undefined reference to `unregister_websocket_types()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: editor/libeditor.windows.opt.tools.64.a(editor_themes.windows.opt.tools.64.o): in function `editor_generate_icon(int, bool, float, bool)':\r\nC:\\Dev\\Godot\\godot.git/editor/editor_themes.cpp:92: undefined reference to `ImageLoaderSVG::create_image_from_string(Ref<Image>, char const*, float, bool, bool)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: editor/libeditor.windows.opt.tools.64.a(editor_themes.windows.opt.tools.64.o): in function `editor_register_and_generate_icons(Ref<Theme>, bool, int, bool)':\r\nC:\\Dev\\Godot\\godot.git/editor/editor_themes.cpp:208: undefined reference to `ImageLoaderSVG::set_convert_colors(Dictionary*)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/editor_themes.cpp:241: undefined reference to `ImageLoaderSVG::set_convert_colors(Dictionary*)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: editor/libeditor.windows.opt.tools.64.a(editor_scene_importer_gltf.windows.opt.tools.64.o): in function `EditorSceneImporterGLTF::_sanitize_scene_name(String const&)':\r\nC:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:159: undefined reference to `RegEx::RegEx(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:160: undefined reference to `RegEx::sub(String const&, String const&, bool, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:159: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:159: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: editor/libeditor.windows.opt.tools.64.a(editor_scene_importer_gltf.windows.opt.tools.64.o): in function `EditorSceneImporterGLTF::_sanitize_bone_name(String const&)':\r\nC:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:190: undefined reference to `RegEx::RegEx(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:191: undefined reference to `RegEx::sub(String const&, String const&, bool, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:193: undefined reference to `RegEx::RegEx(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:194: undefined reference to `RegEx::sub(String const&, String const&, bool, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:196: undefined reference to `RegEx::RegEx(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:197: undefined reference to `RegEx::sub(String const&, String const&, bool, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:199: undefined reference to `RegEx::RegEx(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:200: undefined reference to `RegEx::sub(String const&, String const&, bool, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:199: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:196: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:193: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:190: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:199: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: editor/libeditor.windows.opt.tools.64.a(editor_scene_importer_gltf.windows.opt.tools.64.o):C:\\Dev\\Godot\\godot.git/editor/import/editor_scene_importer_gltf.cpp:196: more undefined references to `RegEx::~RegEx()' follow\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: editor/libeditor.windows.opt.tools.64.a(rename_dialog.windows.opt.tools.64.o): in function `RenameDialog::_regex(String const&, String const&, String const&)':\r\nC:\\Dev\\Godot\\godot.git/editor/rename_dialog.cpp:516: undefined reference to `RegEx::RegEx(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/rename_dialog.cpp:518: undefined reference to `RegEx::sub(String const&, String const&, bool, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/rename_dialog.cpp:516: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/rename_dialog.cpp:516: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: editor/libeditor.windows.opt.tools.64.a(rename_dialog.windows.opt.tools.64.o): in function `RenameDialog::_postprocess(String const&)':\r\nC:\\Dev\\Godot\\godot.git/editor/rename_dialog.cpp:536: undefined reference to `RegEx::RegEx(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/rename_dialog.cpp:537: undefined reference to `RegEx::search_all(String const&, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/rename_dialog.cpp:545: undefined reference to `RegExMatch::get_start(Variant const&) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/rename_dialog.cpp:536: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/editor/rename_dialog.cpp:536: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: scene/libscene.windows.opt.tools.64.a(rich_text_label.windows.opt.tools.64.o): in function `RichTextLabel::parse_expressions_for_values(Vector<String>)':\r\nC:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2855: undefined reference to `RegEx::RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2856: undefined reference to `RegEx::compile(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2857: undefined reference to `RegEx::RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2858: undefined reference to `RegEx::compile(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2859: undefined reference to `RegEx::RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2860: undefined reference to `RegEx::compile(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2861: undefined reference to `RegEx::RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2862: undefined reference to `RegEx::compile(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2863: undefined reference to `RegEx::RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2864: undefined reference to `RegEx::compile(String const&)'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2867: undefined reference to `RegEx::search(String const&, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2863: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2861: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2859: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2857: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2855: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2869: undefined reference to `RegEx::search(String const&, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2874: undefined reference to `RegEx::search(String const&, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2880: undefined reference to `RegEx::search(String const&, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2882: undefined reference to `RegEx::search(String const&, int, int) const'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2863: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2861: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2859: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2857: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: C:\\Dev\\Godot\\godot.git/scene/gui/rich_text_label.cpp:2855: undefined reference to `RegEx::~RegEx()'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::MD5Context::~MD5Context()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:46: undefined reference to `mbedtls_md5_free'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::MD5Context::start()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:51: undefined reference to `mbedtls_md5_starts_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::MD5Context::update(unsigned char const*, unsigned long long)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:56: undefined reference to `mbedtls_md5_update_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::MD5Context::finish(unsigned char*)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:61: undefined reference to `mbedtls_md5_finish_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::SHA1Context::~SHA1Context()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:72: undefined reference to `mbedtls_sha1_free'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::SHA1Context::start()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:77: undefined reference to `mbedtls_sha1_starts_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::SHA1Context::update(unsigned char const*, unsigned long long)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:82: undefined reference to `mbedtls_sha1_update_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::SHA1Context::finish(unsigned char*)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:87: undefined reference to `mbedtls_sha1_finish_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::SHA256Context::~SHA256Context()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:98: undefined reference to `mbedtls_sha256_free'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::SHA256Context::start()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:103: undefined reference to `mbedtls_sha256_starts_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::SHA256Context::update(unsigned char const*, unsigned long long)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:108: undefined reference to `mbedtls_sha256_update_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::SHA256Context::finish(unsigned char*)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:113: undefined reference to `mbedtls_sha256_finish_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::AESContext::~AESContext()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:124: undefined reference to `mbedtls_aes_free'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::AESContext::set_encode_key(unsigned char const*, unsigned long long)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:129: undefined reference to `mbedtls_aes_setkey_enc'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::AESContext::set_decode_key(unsigned char const*, unsigned long long)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:134: undefined reference to `mbedtls_aes_setkey_dec'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::AESContext::encrypt_ecb(unsigned char const*, unsigned char*)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:139: undefined reference to `mbedtls_aes_crypt_ecb'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::AESContext::decrypt_ecb(unsigned char const*, unsigned char*)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:144: undefined reference to `mbedtls_aes_crypt_ecb'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::b64_encode(unsigned char*, int, unsigned long long*, unsigned char const*, int)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:161: undefined reference to `mbedtls_base64_encode'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::b64_decode(unsigned char*, int, unsigned long long*, unsigned char const*, int)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:166: undefined reference to `mbedtls_base64_decode'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::md5(unsigned char const*, int, unsigned char*)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:171: undefined reference to `mbedtls_md5_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::sha1(unsigned char const*, int, unsigned char*)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:176: undefined reference to `mbedtls_sha1_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::sha256(unsigned char const*, int, unsigned char*)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:181: undefined reference to `mbedtls_sha256_ret'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::b64_encode(unsigned char*, int, unsigned long long*, unsigned char const*, int)':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:161: undefined reference to `mbedtls_base64_encode'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::MD5Context::MD5Context()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:42: undefined reference to `mbedtls_md5_init'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::SHA1Context::SHA1Context()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:68: undefined reference to `mbedtls_sha1_init'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::SHA256Context::SHA256Context()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:94: undefined reference to `mbedtls_sha256_init'\r\nC:/Dev/MSYS2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/9.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: core/libcore.windows.opt.tools.64.a(crypto_core.windows.opt.tools.64.o): in function `CryptoCore::AESContext::AESContext()':\r\nC:\\Dev\\Godot\\godot.git/core/crypto/crypto_core.cpp:120: undefined reference to `mbedtls_aes_init'\r\ncollect2.exe: error: ld returned 1 exit status\n"
=====
scons: *** [bin/godot.windows.opt.tools.64.exe] Error 1
```

--- a/SConstruct
+++ b/SConstruct
@@ -68,8 +68,6 @@ env_base.AppendENVPath('PATH', os.getenv('PATH'))
 env_base.AppendENVPath('PKG_CONFIG_PATH', os.getenv('PKG_CONFIG_PATH'))
 env_base.disabled_modules = []
 env_base.use_ptrcall = False
-env_base.split_drivers = False
-env_base.split_modules = False
 env_base.module_version_string = ""
 env_base.msvc = False
 
@@ -129,6 +127,7 @@ opts.Add(BoolVariable('dev', "If yes, alias for verbose=yes warnings=extra werro
 opts.Add('extra_suffix', "Custom extra suffix added to the base filename of all generated binary files", '')
 opts.Add(BoolVariable('vsproj', "Generate a Visual Studio solution", False))
 opts.Add(EnumVariable('macports_clang', "Build using Clang from MacPorts", 'no', ('no', '5.0', 'devel')))
+opts.Add(BoolVariable('split_libmodules', "Split intermediate libmodules.a in smaller chunks to prevent exceeding linker command line size (forced to True when using MinGW)", False))
 opts.Add(BoolVariable('disable_3d', "Disable 3D nodes for a smaller executable", False))
 opts.Add(BoolVariable('disable_advanced_gui', "Disable advanced GUI nodes and behaviors", False))
 opts.Add(BoolVariable('no_editor_splash', "Don't use the custom splash screen for the editor", False))

--- a/drivers/SCsub
+++ b/drivers/SCsub
@@ -46,9 +46,7 @@ if env['vsproj']:
     env.AddToVSProject(env.drivers_sources)
     os.chdir(path)
 
-if env.split_drivers:
-    env.split_lib("drivers")
-else:
-    env.add_source_files(env.drivers_sources, "*.cpp")
-    lib = env.add_library("drivers", env.drivers_sources)
-    env.Prepend(LIBS=[lib])
+env.add_source_files(env.drivers_sources, "*.cpp")
+
+lib = env.add_library("drivers", env.drivers_sources)
+env.Prepend(LIBS=[lib])

--- a/methods.py
+++ b/methods.py
@@ -307,7 +307,7 @@ def split_lib(self, libname, src_list = None, env_lib = None):
         else:
             fname = env.File(f)[0].path
         fname = fname.replace("\\", "/")
-        base = string.join(fname.split("/")[:2], "/")
+        base = "/".join(fname.split("/")[:2])
         if base != cur_base and len(list) > max_src:
             if num > 0:
                 lib = env_lib.add_library(libname + str(num), list)
@@ -320,18 +320,30 @@ def split_lib(self, libname, src_list = None, env_lib = None):
     lib = env_lib.add_library(libname + str(num), list)
     lib_list.append(lib)
 
-    if len(lib_list) > 0:
-        if os.name == 'posix' and sys.platform == 'msys':
-            env.Replace(ARFLAGS=['rcsT'])
-            lib = env_lib.add_library(libname + "_collated", lib_list)
-            lib_list = [lib]
-
     lib_base = []
     env_lib.add_source_files(lib_base, "*.cpp")
     lib = env_lib.add_library(libname, lib_base)
     lib_list.insert(0, lib)
 
     env.Prepend(LIBS=lib_list)
+
+    # When we split modules into arbitrary chunks, we end up with linking issues
+    # due to symbol dependencies split over several libs, which may not be linked
+    # in the required order. We use --start-group and --end-group to tell the
+    # linker that those archives should be searched repeatedly to resolve all
+    # undefined references.
+    # As SCons doesn't give us much control over how inserting libs in LIBS
+    # impacts the linker call, we need to hack our way into the linking commands
+    # LINKCOM and SHLINKCOM to set those flags.
+
+    if '-Wl,--start-group' in env['LINKCOM'] and '-Wl,--start-group' in env['SHLINKCOM']:
+        # Already added by a previous call, skip.
+        return
+
+    env['LINKCOM'] = str(env['LINKCOM']).replace('$_LIBFLAGS',
+            '-Wl,--start-group $_LIBFLAGS -Wl,--end-group')
+    env['SHLINKCOM'] = str(env['LINKCOM']).replace('$_LIBFLAGS',
+            '-Wl,--start-group $_LIBFLAGS -Wl,--end-group')
 
 
 def save_active_platforms(apnames, ap):

--- a/modules/SCsub
+++ b/modules/SCsub
@@ -16,7 +16,7 @@ for x in env.module_list:
     env_modules.Append(CPPDEFINES=["MODULE_" + x.upper() + "_ENABLED"])
     SConscript(x + "/SCsub")
 
-if env.split_modules:
+if env['split_libmodules']:
     env.split_lib("modules", env_lib = env_modules)
 else:
     lib = env_modules.add_library("modules", env.modules_sources)

--- a/platform/windows/detect.py
+++ b/platform/windows/detect.py
@@ -151,14 +151,14 @@ def setup_msvc_auto(env):
         env['bits'] = '64'
     else:
         env['bits'] = '32'
-    print(" Found MSVC version %s, arch %s, bits=%s" % (env['MSVC_VERSION'], env['TARGET_ARCH'], env['bits']))
+    print("Found MSVC version %s, arch %s, bits=%s" % (env['MSVC_VERSION'], env['TARGET_ARCH'], env['bits']))
     if env['TARGET_ARCH'] in ('amd64', 'x86_64'):
         env["x86_libtheora_opt_vc"] = False
 
 def setup_mingw(env):
     """Set up env for use with mingw"""
     # Nothing to do here
-    print("Using Mingw")
+    print("Using MinGW")
     pass
 
 def configure_msvc(env, manual_msvc_config):
@@ -294,7 +294,10 @@ def configure_mingw(env):
     ## Compiler configuration
 
     if (os.name == "nt"):
-        env['ENV']['TMP'] = os.environ['TMP']  # way to go scons, you can be so stupid sometimes
+        # Force splitting libmodules.a in multiple chunks to work around
+        # issues reaching the linker command line size limit, which also
+        # seem to induce huge slowdown for 'ar' (GH-30892).
+        env['split_libmodules'] = True
     else:
         env["PROGSUFFIX"] = env["PROGSUFFIX"] + ".exe"  # for linux cross-compilation
 

